src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5248,66 +5252,93 @@ static BoundNode bindElement(CollectionElementSyntax syntax, BindingDiagnosticBa
+            var expressionPlaceholder = new BoundCollectionExpressionSpreadExpressionPlaceholder(syntax.Expression, expression.Type);
+            var enumeratorInfo = builder.Build(location: default);
+            var collectionType = enumeratorInfo.CollectionType;
+            var useSiteInfo = this.GetNewCompoundUseSiteInfo(diagnostics);
+            var conversion = this.Conversions.ClassifyConversionFromExpression(expression, collectionType, isChecked: this.CheckOverflowAtRuntime, ref useSiteInfo);
+            Debug.Assert(conversion.IsValid);
+            diagnostics.Add(syntax.Expression, useSiteInfo);
+            var convertedExpression = this.ConvertForEachCollection(expressionPlaceholder, conversion, collectionType, diagnostics);
 
-                BoundExpression? lengthOrCount;
+            BoundExpression? lengthOrCount;
 
-                if (enumeratorInfo is { InlineArraySpanType: not WellKnownType.Unknown })
-                {
-                    _ = expression.Type.HasInlineArrayAttribute(out int length);
-                    Debug.Assert(length > 0);
-                    lengthOrCount = new BoundLiteral(expression.Syntax, ConstantValue.Create(length), @this.GetSpecialType(SpecialType.System_Int32, diagnostics, expression.Syntax)) { WasCompilerGenerated = true };
-                }
-                else if (!@this.TryBindLengthOrCount(syntax.Expression, expressionPlaceholder, out lengthOrCount, diagnostics))
+            if (enumeratorInfo is { InlineArraySpanType: not WellKnownType.Unknown })
+            {
+                _ = expression.Type.HasInlineArrayAttribute(out int length);
+                Debug.Assert(length > 0);
+                lengthOrCount = new BoundLiteral(expression.Syntax, ConstantValue.Create(length), this.GetSpecialType(SpecialType.System_Int32, diagnostics, expression.Syntax)) { WasCompilerGenerated = true };
+            }
+            else if (!this.TryBindLengthOrCount(syntax.Expression, expressionPlaceholder, out lengthOrCount, diagnostics))
+            {
+                lengthOrCount = null;
+            }
+
+            return new BoundCollectionExpressionSpreadElement(
+                syntax,
+                expression,
+                expressionPlaceholder: expressionPlaceholder,
+                conversion: convertedExpression,
+                enumeratorInfo,
+                lengthOrCount: lengthOrCount,
+                elementPlaceholder: null,
+                iteratorBody: null,
+                hasErrors: false);
+        }
+
+        private BoundNode BindCollectionArguments(WithElementSyntax syntax, BindingDiagnosticBag diagnostics)
+        {
+            MessageID.IDS_FeatureCollectionExpressionArguments.CheckFeatureAvailability(diagnostics, syntax.WithKeyword);
+
+            var arguments = AnalyzedArguments.GetInstance();
+            BindArgumentsAndNames(syntax.ArgumentList, diagnostics, arguments, allowArglist: true);
```

Comment:
>allowArglist: true [](http://example.com/codeflow?start=79&length=18)

Consider adding a test to cover this

Reply:
Good catch, thanks, I've added tests.

That said, we should consider not supporting `__arglist`.

Reply:
Added an open question in https://github.com/dotnet/csharplang/pull/9115.